### PR TITLE
Feature/#2965 add support to column tab and field in packout

### LINKED ISSUE #2965: Add support to Column, Tab and Field in Packout
## Feature request
In large windows and tables with few changes, large xml remain with few real corrections.

#### Describe the solution you'd like

Add support to Column, Tab and Fields objects in the packout detail.

image

image

image

#### Additional context

Also add 3 Smart Browsers to add element from "Window" tab, "Tab" tab or "Table" tab direct to the Packout (in the gear), in each Smart Browser select the items to be exported and in the process select the destination package (the SB remember which was the last selected Packout)

image

image

image

Any suggestion is welcome

@yamelsenih, @marcalwestf , @e-Evolution

### PR DESCRIPTION
**Pull request by rcapecce**
Fixes #2965

### CODE REVIEW

**Comment by marcalwestf** [comment]
Hi @rcapecce  it seems the XML file has to be extracted from _feature_2965.zip_.
See image
Selection_316

**Comment by rcapecce** [comment]
That's right @marcalwestf , the functionality dictionary is in the feature_2965.zip file, it's a packout, it's not an XML migration file.

File: base/src/org/adempiere/pipo/PackOut.java [hunk 1]
```diff
@@ -1,973 +1,974 @@
...
+		
+		return "Finish Process";
+	}	//	doIt
+	
+	private void initContext() {
+		Properties tmp = new Properties();
+		if (getCtx() != null)
+			tmp.putAll(getCtx());
+		tmp.put("TrxName", get_TrxName());
+		tmp.put("PackOutPro
```
Comment (yamelsenih) [inline base/src/org/adempiere/pipo/PackOut.java]: This class have many changes, all class is replaced, what is the reason?

File: base/src/org/compiere/model/X_AD_Package_Exp_Detail.java [hunk 2]
```diff
@@ -99,9 +99,37 @@ public int getAD_Browse_ID ()
...
+	public I_AD_Column getAD_Column() throws RuntimeException
     {
-		return (org.compiere.model.I_AD_EntityType)MTable.get(getCtx(), org.compiere.model.I_AD_EntityType.Table_Name)
+		return (I_AD_Column)MTable.get(getCtx(), I_AD_Column.Table_Name)
+			.getPO(getAD_Column_ID(), get_TrxName());	}
+
+	/** Set Column.
+		@param AD_Column_ID 
+		Column in the table
+	  */
```
Comment (yamelsenih) [inline base/src/org/compiere/model/X_AD_Package_Exp_Detail.java]: This class why is changed?

File: base/src/org/compiere/model/X_AD_Package_Exp_Detail.java [hunk 3]
```diff
@@ -127,9 +155,37 @@ public int getAD_EntityType_ID ()
...
+
+	/** Set Field.
+		@param AD_Field_ID 
+		Field on a database table
+	  */
+	public void setAD_Field_ID (int AD_Field_ID)
+	{
+		if (AD_Field_ID < 1) 
+			set_Value (COLUMNNAME_AD_Field_ID, null);
+		else
```
Comment (yamelsenih) [inline base/src/org/compiere/model/X_AD_Package_Exp_Detail.java]: I don't sure if this change is necessary

File: migration/393lts-394lts/04410_AddOpnUp_EntityType.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<Migrations>
+  <Migration EntityType="D" Name="OpenUp Entity Type" ReleaseNo="3.9.1" SeqNo="4410">
```
Comment (yamelsenih) [inline migration/393lts-394lts/04410_AddOpnUp_EntityType.xml]: 3.9.3 version instead

**Comment by marcalwestf** [comment]
hi @rcapecce  now you should have received login + password for the central IDs, it would be possible you create and push the XML file.

**Comment by marcalwestf** [comment]
@rcapecce  have you been able to move on this issue?
Thanks

**Comment by yamelsenih** [comment]
Hi @rcapecce maybe this PR can be rebuilding with a fresh changes from develop, can you make it for review? If is not possible just let me know for close meawhile

**Comment by rcapecce** [comment]
Hi @yamelsenih, this issue can be closed because we had a different approach.
Now that we have a little more experience in packouts, I see that it is better to use a packout for the entire window instead of for specific fields.

**Pull request closed without merging**